File: src/main/java/com/github/frcsty/litebansdiscord/DiscordPlugin.java
package com.github.frcsty.litebansdiscord;

import com.github.frcsty.litebansdiscord.discord.Discord;
import com.github.frcsty.litebansdiscord.discord.command.CheckBanCommand;
import com.github.frcsty.litebansdiscord.discord.command.HistoryCommand;
import com.github.frcsty.litebansdiscord.discord.command.IpHistoryCommand;
import me.mattstudios.mfjda.base.CommandManager;
import org.bukkit.plugin.java.JavaPlugin;

import java.util.Arrays;

public final class DiscordPlugin extends JavaPlugin {

    private final Discord discord = new Discord(this);

    @Override
    public void onEnable() {
        saveDefaultConfig();

        final CommandManager commandManager = new CommandManager(discord.getJda());

        commandManager.register(Arrays.asList(
                new CheckBanCommand(this),
                new HistoryCommand(this),
                new IpHistoryCommand(this)
        ));
    }

    @Override
    public void onDisable() {
        reloadConfig();
    }

}

File: src/main/kotlin/com/github/frcsty/litebansdiscord/DiscordPlugin.kt
package com.github.frcsty.litebansdiscord

import com.github.frcsty.litebansdiscord.discord.Discord
import org.bukkit.plugin.java.JavaPlugin
import java.util.logging.Level

class DiscordPlugin : JavaPlugin() {

    val prefix = "-"

    override fun onEnable() {
        saveDefaultConfig()

        Discord(this)
    }

    override fun onDisable() {
        reloadConfig()
    }
}

fun log(text: String) {
    val plugin = JavaPlugin.getProvidingPlugin(DiscordPlugin::class.java)

    plugin.logger.log(Level.INFO, text)
}